src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -140,21 +142,20 @@ class LibraryService (protected val argUserInfo: UserInfo,
     }
   }
 
+  // should only be used to change published state
   def setWorkspaceIsPublished(ns: String, name: String, value: Boolean): Future[PerRequestMessage] = {
     rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
       val pub = isPublished(workspaceResponse)
-      val requiredLevel = if (!pub || (pub && !value)) WorkspaceAccessLevels.Owner else WorkspaceAccessLevels.Write
-      if (workspaceResponse.accessLevel >= requiredLevel) {
-        val operations = updatePublishAttribute(value)
-        rawlsDAO.patchWorkspaceAttributes(ns, name, operations) map { ws =>
+      if (pub == value)
+        Future(RequestComplete(NoContent))
```

Comment:
the re-publish (i.e. when library attributes have been modified for a published dataset) happens in the method "internalPatchWorkspaceAndRepublish". the method "setWorkspaceIsPublished" should only be invoked when changing the published state. i think the reason we have these 2 methods is that the api treats them separately and when we were checking permissions in orch, they had different requirements